libs/datamodel/parser-database/src/attributes/default.rs

```diff
@@ -0,0 +1,224 @@
+use crate::{
+    ast::{self, WithName},
+    context::{Arguments, Context},
+    types::{DefaultAttribute, ScalarField, ScalarFieldType, ScalarType},
+    DatamodelError,
+};
+
+/// @default on scalar fields
+pub(super) fn visit_field_default<'ast>(
+    args: &mut Arguments<'ast>,
+    field_data: &mut ScalarField<'ast>,
+    model_id: ast::ModelId,
+    field_id: ast::FieldId,
+    ctx: &mut Context<'ast>,
+) {
+    let value = match args.default_arg("value") {
+        Ok(value) => value,
+        Err(err) => return ctx.push_error(err),
+    };
+
+    let ast_model = &ctx.db.ast[model_id];
+    let ast_field = &ast_model[field_id];
+    if ast_field.arity.is_list() {
+        return ctx.push_error(args.new_attribute_validation_error("Cannot set a default value on list field."));
```

Comment:
We should check whether we already have an issue for removing this. This is a pretty arbitrary legacy limitation.

Reply:
Couldn't find an issue about it, so I made one https://github.com/prisma/prisma/issues/10773

Reply:
:+1: